src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import classNames from 'classnames';
+
+import { TodoLoadingOverlay } from '../TodoLoadingOverlay';
+import { TodoEditForm } from '../TodoEditForm';
+
+import { TodoRich } from '../../types/TodoRich';
+import { TodoMode } from '../../types/TodoMode';
+import { TodoRichEditable } from '../../types/TodoRichEditable';
+
+type Props = {
+  todo: TodoRich;
+  onTodoDelete?: (todoId: number) => Promise<void>;
+  onTodoUpdate?: (todoId: number, updatedData: TodoRichEditable)
+  => Promise<void>;
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo: {
+    id,
+    title,
+    completed,
+    mode,
+  },
+  onTodoDelete = () => {},
+  onTodoUpdate = () => {},
+}) => {
+  const hadndleTodoDelete = async () => onTodoDelete(id);
+
+  const hadndleTodoToggle = async (
+    checkEvent: React.ChangeEvent<HTMLInputElement>,
+  ) => onTodoUpdate(id, { completed: checkEvent.target.checked });
+
+  const handleTodoTitleUpdate = async (newTitle: string) => (
+    onTodoUpdate(id, { title: newTitle.trim() })
+  );
+
+  const handleEditingSkip = async () => (
+    onTodoUpdate(id, { mode: TodoMode.None })
+  );
+
+  const handleEditingStart = async () => (
+    onTodoUpdate(id, { mode: TodoMode.Editing })
+  );
```

Comment:
3 same functions, you can make shrink them in one

Reply:
I believe it's better to leave the existing functions as they are.

**Main point**: I think it helps to prioritize **readability** and adopt these functions for use on **other levels** of the element hierarchy.

1. There is already a generalized function `onTodoUpdate` that can receive specific todo data, and the three functions serve as wrappers that invoke it. Using more specific function names like `Toggle`, `SkipEditing`, and `StartEditing` makes the code more readable.

2. Two of the functions are passed forward to the next component, and it would be more convenient to simply apply `onEditingSkip()` rather than look at the implementation one level up to figure out what parameters should be passed in.

3. One of the three functions `handleEditingStart` is used as a handler in the current component. If I follow your suggestion, I should need to keep this handler as it is + add your suggested generalized function. This would result in reducing the number of functions from 3 to 2, which is not a significant improvement.